force-app/main/default/classes/SOQL.cls

```diff
@@ -85,9 +104,25 @@ public with sharing class SOQL {
      * @return A map of Task Ids to Task records.
      * Update the corresponding test class and testGetTasksDueNextWeek method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
-    //  The body of the method
-    //}
+    public static Map<Id, Task> getTasksDueNextWeek(Date startDate)
+    {
+        // Calculate the end date for 7 days from the start date
+        Date endDate = startDate.addDays(7);
+        // Query tasks where the due date is between the start date and 7 days from that date
+    List<Task> taskList = [SELECT Id, Subject, ActivityDate 
+    FROM Task 
+    WHERE ActivityDate >= :startDate 
+    AND ActivityDate <= :endDate];
+
+    // Create a map of Task Ids to Task records
+    Map<Id, Task> taskMap = new Map<Id, Task>();
+    for (Task t : taskList) 
+    {
+    taskMap.put(t.Id, t);
+    }
```

Comment:
You also return your query directly to a map by putting the soql right in the map constructor `new Map<Id, Task>([QUERY HERE]);`